hyperspy/misc/hyperspy_magics.py

```diff
@@ -14,11 +14,22 @@ def hyperspy(self, line):
 
         Load HyperSpy, numpy and matplotlib to work interactively.
 
-        %hyperspy runs the following commands::
+        %hyperspy runs the following commands in various cases:
 
+        >>> # if toolkit is "None" only
+        >>> import matplotlib
+        >>> matplotlib.use('Agg')
+
+        >>> # if toolkit is "qt4" only
+        >>> import os
+        >>> os.environ['QT_API'] = 'pyqt'
+
+        >>> # if toolkit is not "None"
+        >>> %matplotlib [toolkit]
+
+        >>> # run in all cases
         >>> import numpy as np
         >>> import hyperspy.api as hs
-        >>> %matplotlib [toolkit]
         >>> import matplotlib.pyplot as plt
 
         If you pass `-r`, the current input cell will be overwritten with the above specified commands. As a
```

Comment:
Done by `pylab` example... They have their `--no-import-all` described in both places